### PR TITLE
Create SegmentBar element

### PR DESCRIPTION
**Pull request by mixth**
# Update

According to #66. I have:
- Changed element name to `StackedBarChart`
- Hide label or percentage when there is not enough space
- Added tooltips and made them display when hover on each cell (regardless of hiddenness of its label or percentage)

Special thanks to @Th1nkK1D for his #77 that inspired many parts in this pull request 🥳

## Screenshots
Screen Shot 2019-11-05 at 1 04 42 AM
Screen Shot 2019-11-05 at 1 04 53 AM

-----
# Outdated
~~Prepare for #13~~
~~I found that these segment bars can be separated to its own element which can reduce works required to build those pages.~~

## ~~Screenshot~~
Screen Shot 2019-11-03 at 3 56 46 PM

## ~~How to Use~~
~~I've created the class called `SegmentProp`. And add an example in the Cabinet page.~~
```javascript
export class SegmentProp {
  label = ""
  color = "#ffffff"
  backgroundColor = "#000000"
  volume = 0
}
```
~~`SegmentBar` element consumes an array of this `SegmentProp`.~~

### CODE REVIEW

**Comment by mixth** [comment]
@rapee oh... I didn't see #66. Should I update this to match requirements in it?

**Comment by rapee** [comment]
@mixth Please, and sorry for the additional works. You can keep the naming of the component.
 I just added #66 yesterday.

My only comment is that an array of data would be more appropriate props for SegmentBar. You won't need to SegmentProp when using SegmentBar. For example:

```js
const ages = [
  { name: "15-24 ปี", value: 142, background: "#eeeeee" },
  { name: "25-38 ปี", value: 362, background: "#cccccc" },
  { name: "39-73 ปี", value: 456, background: "#aaaaaa" },
  { name: "74-91 ปี", value: 12, background: "#666666" }
]
```

```jsx
<SegmentBar data={ages}></SegmentBar>
```

**Comment by rapee** [comment]
Hi again. We have a duplicate PR here (#77) (Please excuse my confusion making issues T_T).

Just want to say that you can proceed your work. But you can also learn from what @Th1nkK1D did in that PR :)

For example, he passes props as array.

File: src/components/segmentBar.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        border: "1px solid black",
+        minHeight: 44,
+        fontFamily: "var(--ff-title)",
+        margin: "2rem",
+      }}
+    >
+      {segmentProps.map(segmentProp => (
+        <div
+          css={{
+            width: (segmentProp.volume / totalVolume) * 100 + "%",
```
Comment (rapee) [inline src/components/segmentBar.js]: Regarding the use of `width`, I suggest you to separate it as `style={{ width: ... }}`.

The reason is that when using `css={{}}` it compiles to `class="..."`, while `style={{ }}` compiles to, surprisingly, `style="..."`. Try switching between these two and inspect it for yourself.

That being said, width attribute is kinda dynamic here depending on data, so using "style" is better I think.

Comment (mixth) [reply src/components/segmentBar.js]: Thanks for the review! I have reflected your review in the new commits. I am quite a newbie in React right now. Definitely need to do more homework. 😅

**Comment by mixth** [comment]
@rapee Updated :)

**Comment by rapee** [comment]
@mixth This looks awesome!

**Pull request merged**